lib/web_ui/lib/src/engine/canvaskit/image_web_codecs.dart

```diff
@@ -0,0 +1,436 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+/// Uses the `ImageDecoder` class supplied by the browser.
+///
+/// See also:
+///
+///  * `image_wasm_codecs.dart`, which uses codecs supplied by the CanvasKit WASM bundle.
+@JS()
+library image_web_codecs;
+
+import 'dart:async';
+import 'dart:html' as html;
+import 'dart:js_util' as js_util;
+import 'dart:math' as math;
+import 'dart:typed_data';
+
+import 'package:js/js.dart';
+import 'package:ui/ui.dart' as ui;
+
+import '../browser_detection.dart';
+import '../util.dart';
+import 'canvaskit_api.dart';
+import 'image.dart';
+import 'surface.dart';
+import 'surface_factory.dart';
+
+@JS('window.ImageDecoder')
+external Object? get _imageDecoderConstructor;
+
+/// Whether the current browser supports `ImageDecoder`.
+bool browserSupportsImageDecoder =
+    _imageDecoderConstructor != null && browserEngine == BrowserEngine.blink;
+
+/// Sets the value of [browserSupportsImageDecoder] to its default value.
+void debugResetBrowserSupportsImageDecoder() {
+  browserSupportsImageDecoder =
+      _imageDecoderConstructor != null;
+}
+
+/// Image decoder backed by the browser's `ImageDecoder`.
+class CkBrowserImageDecoder implements ui.Codec {
+  static Future<CkBrowserImageDecoder> create({
+    required Uint8List data,
+    required String debugSource,
+    int? targetWidth,
+    int? targetHeight,
+  }) async {
+    // ImageDecoder does not detect image type automatically. It requires us to
+    // tell it what the image type is.
+    final String? contentType = detectContentType(data);
+
+    if (contentType == null) {
+      final String fileHeader;
+      if (data.isNotEmpty) {
+        fileHeader = '[' + bytesToHexString(data.sublist(0, math.min(10, data.length))) + ']';
+      } else {
+        fileHeader = 'empty';
+      }
+      throw ImageCodecException(
+        'Failed to detect image file format using the file header.\n'
+        'File header was $fileHeader.\n'
+        'Image source: $debugSource'
+      );
+    }
+
+    final _ImageDecoder webDecoder = _ImageDecoder(_ImageDecoderOptions(
```

Comment:
I think you want this in the try/catch below.

Reply:
Done.